File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Views/ViewCourseRegister.cs
﻿using KnowledgeSharingApi.Domains.Models.Entities.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Views
{
    [Table("ViewCourseRegister")]
    public class ViewCourseRegister : CourseRegister
    {
        public string Username { get; set; } = string.Empty;

        public string FullName { get; set; } = string.Empty;

        public string? Avatar { get; set; }

        public string? Cover { get; set; }

        public string? Thumbnail { get; set; }

        public decimal? Fee { get; set; }

        public bool? IsFree { get; set; }

        public string Title { get; set; } = string.Empty;

        public string? Abstract { get; set; } = string.Empty;

        public Guid CourseOwnerUserId { get; set; }

        public string CourseOwnerUsername { get; set; } = string.Empty;
        
        public string CourseOwnerFullName { get; set; } = string.Empty;

        public string? CourseOwnerAvatar { get; set; }

        public string? CourseOwnerCover { get; set; }

        protected override ViewCourseRegister Init()
        {
            return new ViewCourseRegister();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/views/view-course-register.js
// Framework: .NET
// Technology stack: C#, Entity Framework
import CourseRegister from '../entities/course-register';

class ViewCourseRegister extends CourseRegister {
    constructor() {
        super();
        this.Username = null;
        this.FullName = null;
        this.Avatar = null;
        this.Cover = null;
        this.Thumbnail = null;
        this.Title = null;
        this.Abstract = null;
        this.CourseOwnerUserId = null;
        this.CourseOwnerUsername = null;
        this.CourseOwnerFullName = null;
        this.CourseOwnerAvatar = null;
        this.CourseOwnerCover = null;
    }

    init() {
        return new ViewCourseRegister();
    }

    getCourse(){
        return {
            UserItemId: this.CourseId,
            Title: this.Title,
            Abstract: this.Abstract,
            Thumbnail: this.Thumbnail
        }
    }

    getOwner(){
        return {
            UserId: this.CourseOwnerUserId,
            Username: this.CourseOwnerUsername,
            FullName: this.CourseOwnerFullName,
            Avatar: this.CourseOwnerAvatar,
            Cover: this.CourseOwnerCover
        }
    }
}

export default ViewCourseRegister;
